modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/latch/ExchangeLatchManager.java

```diff
@@ -362,6 +395,38 @@ public boolean canSkipJoiningNodes(AffinityTopologyVersion topVer) {
             .allMatch(node -> node.version().compareTo(PROTOCOL_V2_VERSION_SINCE) >= 0);
     }
 
+    /**
+     * Processes ack message immediately or delay processing until local latch will be created.
+     *
+     * @param from Node sent ack.
+     * @param message Ack message.
+     */
+    private void processAck(UUID from, LatchAckMessage message) {
+        lock.lock();
+
+        try {
+            if (!isLocalLatchCreated(message.topVer())) {
+                awaitedLatchVersions.computeIfAbsent(message.topVer(), (k) -> new GridFutureAdapter<>())
```

Comment:
The same issue with several latches here. We should take LatchAckMessage#latchId into account.
We can write a test with second latch using PartitionsExchangeAware mechanism.